portafly/src/tests/components/shared/data-list/DataListTable.test.tsx

```diff
@@ -0,0 +1,304 @@
+    expect(checkboxes.filter(isChecked)).toHaveLength(1)
+    // TODO: assert text '1 Selected' in BulkSelectorWidget test
+
+    const row = wrapper.getRow(selectedCharacter)
+    fireEvent.click(within(row).getByRole('checkbox'))
+
+    expect(defaultTable.selectOne).toHaveBeenLastCalledWith(selectedCharacter.id, false)
+  })
+
+  it('should remove all selections when clicking the checkbox', () => {
+    const { bulkSelectorDropdown: dropdown } = setup()
+
+    fireEvent.click(dropdown.getByRole('checkbox'))
+
+    expect(defaultTable.selectAll).toHaveBeenLastCalledWith(false, expect.any(Array))
+  })
+
+  it('should be able to deselect all', () => {
+    const { bulkSelectorDropdown: dropdown } = setup()
+
+    fireEvent.click(dropdown.getByRole('button'))
+    fireEvent.click(dropdown.getByText('bulk_selector.none'))
+
+    expect(defaultTable.selectAll).toHaveBeenLastCalledWith(false)
+  })
+
+  it('should display bulk actions enabled and no warning', () => {
+    const { bulkActionsDropdown: dropdown } = setup()
+
+    fireEvent.click(dropdown.getByRole('button'))
+
+    expect(dropdown.queryByText('bulk_actions.warning')).not.toBeInTheDocument()
+    expect(dropdown.getByText('Send Raven')).toBeEnabled()
+    expect(dropdown.getByText('Change State')).toBeEnabled()
+  })
+})
+
+describe('when there are any records, and all are selected', () => {
+  beforeEach(() => {
+    useTableMock.mockReturnValue({
+      ...defaultTable,
+      rows: rows.map((r) => ({ ...r, selected: true })),
+      selectedRows: rows
+    })
+  })
+
+  it('should render selected records and be able to deselect them', () => {
+    const wrapper = setup()
+    const checkboxes = wrapper.getAllByRole('checkbox')
+    expect(checkboxes).toHaveLength(characters.length + 1)
+    expect(checkboxes.every(isChecked)).toBe(true)
+    // TODO: assert text '2 Selected' in BulkSelectorWidget test
+
+    const character = characters[0]
+    const row = wrapper.getRow(character)
+    fireEvent.click(within(row).getByRole('checkbox'))
+
+    expect(defaultTable.selectOne).toHaveBeenLastCalledWith(character.id, false)
+  })
+
+  it('should remove all selections when clicking the checkbox', () => {
+    const { bulkSelectorDropdown: dropdown } = setup()
+
+    fireEvent.click(dropdown.getByRole('checkbox'))
+
+    expect(defaultTable.selectAll).toHaveBeenLastCalledWith(false, expect.any(Array))
+  })
+
+  it('should be able to deselect all', () => {
+    const { bulkSelectorDropdown: dropdown } = setup()
+
+    fireEvent.click(dropdown.getByRole('button'))
+    fireEvent.click(dropdown.getByText('bulk_selector.none'))
+
+    expect(defaultTable.selectAll).toHaveBeenLastCalledWith(false)
+  })
+})
+
+describe('when there are any filters', () => {
+  const filteredCharacter = characters[0]
+  const hiddenCharacter = characters[1]
+
+  beforeEach(() => {
+    useFiltersMock.mockReturnValue({
+      filters: {
+        state: ['dead']
```

Comment:
Apart from the spoiler, perhaps use a less strong/dark term? (it's a public repo, IDK)
```suggestion
        state: ['deceased']
```